web/src/narrow_state.ts

```diff
@@ -33,7 +32,7 @@ export function filter(): Filter | undefined {
 
 export function search_terms(): Term[] {
     if (current_filter === undefined) {
-        return new Filter(page_params.narrow).terms();
+        return [];
```

Comment:
This probably breaks the `page_params.narrow` feature -- which is intended to let you pass a narrow via the `GET` parameters and have that view be the default view.

I am potentially OK with breaking that feature temporarily -- it's not a thing we really document or officially support, and probably a cleaner implementation will be available after we complete this migration. But do we need to do that? I think we could just add an extra if/else clause to use `page_params.narrow` here if set, and otherwise fall back to `[]`.

Reply:
Yeah, did that.

Reply:
I guess I'm just generally not sure if we should be returning `[]` or the search terms for `home_msg_list` here.

Reply:
I guess it makes sense to return `in:home` since if `current_filter` is undefined, user is either in `All messages`, Recent conversations or `Inbox`.

Reply:
Updated with that and added another commit for the `get_users_typing_for_narrow` fix.